src/StatusQ/Controls/StatusBanner.qml

```diff
@@ -1,31 +1,71 @@
 import QtQuick 2.14
 
 import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
 
 Column {
     id: statusBanner
     width: parent.width
 
-    property color backgroundColor
-    property color bordersColor
-    property color fontColor
     property string statusText
+    property int type: StatusBanner.Type.Info
     property int textPixels: 15
     property int statusBannerHeight: 38
 
+    // "private" properties
+    QtObject {
+           id: pProp
+           property color backgroundColor
+           property color bordersColor
+           property color fontColor
+    }
+
+    enum Type {
+        None, // 0
+        Info, // 1
+        Danger, // 2
+        Success, // 3
+        Warning // 4
+    }
+
+    function setBannerBackgroundColor() {
```

Comment:
@noeliaSD could you please implement this using states? it makes more sense and will look cleaner having corresponding states to type changes. also I see all property changes are happening in private component's properties.